data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/source/RandomStringSource.java

```diff
@@ -38,56 +34,35 @@
     static final String EVENT_TYPE = "event";
     private static final Logger LOG = LoggerFactory.getLogger(RandomStringSource.class);
 
-    private ExecutorService executorService;
     private volatile boolean stop = false;
 
-    private void setExecutorService() {
-        if(executorService == null || executorService.isShutdown()) {
-            executorService = Executors.newSingleThreadExecutor(
-                    new ThreadFactoryBuilder().setDaemon(false).setNameFormat("random-source-pool-%d").build()
-            );
-        }
-    }
-
     @Override
     public void start(final Buffer<Record<Event>> buffer) {
-        setExecutorService();
-        executorService.execute(() -> {
-            while (!stop) {
-                try {
-                    LOG.info("Writing to buffer");
-                    final Record<Event> record = generateRandomStringEventRecord();
-                    buffer.write(record, 500);
-                    Thread.sleep(500);
-                } catch (final InterruptedException e) {
-                    break;
-                } catch (final TimeoutException e) {
-                    // Do nothing
-                }
+        while (!stop) {
```

Comment:
I don't think this is right. `Pipeline` calls `Source::start` on start-up, on the main thread. Won't this block Data Prepper?

Reply:
So you are saying the executor service is required here then?

Reply:
Yes. The processor thread calls `start` and expects it to return. So the random data needs to be generated in a different thread.

Reply:
I was thinking that the workers would still be able to pull from the buffer, but you're right start does get called before any processors are started. Thanks for explaining.